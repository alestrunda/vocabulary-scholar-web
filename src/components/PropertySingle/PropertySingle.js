import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const PropertySingle = ({ align, children, className, label, style }) => (
  <div
    className={classNames(
      'property-single',
      { 'property-single--left': align === 'left' },
      { 'property-single--right': align === 'right' },
      className
    )}
    style={style}
  >
    <span className="property-single__value">{children}</span>
    <span className="property-single__label">{label}</span>
  </div>
);

PropertySingle.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string])
    .isRequired,
  label: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
  className: PropTypes.string,
  align: PropTypes.string,
  style: PropTypes.object,
};

export default PropertySingle;
