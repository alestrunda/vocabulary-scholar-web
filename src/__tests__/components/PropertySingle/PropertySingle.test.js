import React from 'react';
import { shallow } from 'enzyme';

import PropertySingle from '../../../components/PropertySingle';

it('renders correctly', () => {
  const detail = 'detail';
  const content = <span>content</span>;
  const root = shallow(
    <PropertySingle detail={detail}>{content}</PropertySingle>
  );
  expect(root).toMatchSnapshot();
});
