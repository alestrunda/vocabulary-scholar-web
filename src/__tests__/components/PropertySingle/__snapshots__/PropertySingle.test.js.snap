// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PropertySingle
    detail="detail"
  >
    <span>
      content
    </span>
  </PropertySingle>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="property-single__value"
        >
          <span>
            content
          </span>
        </span>,
        <span
          className="property-single__label"
        />,
      ],
      "className": "property-single",
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            content
          </span>,
          "className": "property-single__value",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "content",
          },
          "ref": null,
          "rendered": "content",
          "type": "span",
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "property-single__label",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="property-single__value"
          >
            <span>
              content
            </span>
          </span>,
          <span
            className="property-single__label"
          />,
        ],
        "className": "property-single",
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              content
            </span>,
            "className": "property-single__value",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "content",
            },
            "ref": null,
            "rendered": "content",
            "type": "span",
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "property-single__label",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
